### PR TITLE
[CS] Code style Tabs must be used to indent lines; round 2

### PR DESCRIPTION
**Pull request by photodude**
Pull Request for Issue Tabs must be used to indent lines; spaces are not allowed.

### Summary of Changes
- Tabs must be used to indent lines; spaces are not allowed

Working through Automatic fixers to remove cases where we have had rule exceptions

[Automatically fixed with Joomla code standards 2.0.0 PHPCS2-alpha2 fixers](https://github.com/joomla/coding-standards/releases/tag/2.0.0-alpha2)

None of the manual only fixes have been applied

### Testing Instructions
Merge by code review

### Expected result
code style has been applied as listed above, old code style testing on drone does not error.

### Actual result
code style had not been applied. [Autofixers from the Joomla code standards 2.0.0 PHPCS2 alpha2](https://github.com/joomla/coding-standards/releases/tag/2.0.0-alpha2) were used to implement fixable code style

### Documentation Changes Required
none

### CODE REVIEW

File: templates/beez3/html/com_contact/category/default_children.php [hunk 1]
```diff
@@ -32,15 +32,15 @@
...
-            <?php endif; ?>
+			<?php endif; ?>
 
-            <?php if ($this->params->get('show_cat_items') == 1) :?>
+			<?php if ($this->params->get('show_cat_items') == 1) :?>
 			<dl><dt>
 				<?php echo JText::_('COM_CONTACT_CAT_NUM'); ?></dt>
 				<dd><?php echo $child->numitems; ?></dd>
 			</dl>
 		<?php endif; ?>
```
Comment (Quy) [inline templates/beez3/html/com_contact/category/default_children.php]: Indent

File: templates/beez3/html/com_content/category/blog_item.php [hunk 2]
```diff
@@ -116,9 +116,9 @@
 		</dd>
 <?php endif; ?>
 <?php if ($params->get('show_author') or $params->get('show_category') or $params->get('show_create_date') or $params->get('show_modify_date') or $params->get('show_publish_date') or $params->get('show_parent_category') or $params->get('show_hits')) :?>
- 	</dl>
+	</dl>
 <?php endif; ?>
-<?php  if (isset($images->image_intro) and !empty($images->image_intro)) : ?>
+<?php if (isset($images->image_intro) and !empty($images->image_intro)) : ?>
```
Comment (Quy) [inline templates/beez3/html/com_content/category/blog_item.php]: change `and`

File: templates/beez3/html/com_content/category/blog_item.php [hunk 3]
```diff
@@ -116,9 +116,9 @@
 		</dd>
 <?php endif; ?>
 <?php if ($params->get('show_author') or $params->get('show_category') or $params->get('show_create_date') or $params->get('show_modify_date') or $params->get('show_publish_date') or $params->get('show_parent_category') or $params->get('show_hits')) :?>
```
Comment (Quy) [inline templates/beez3/html/com_content/category/blog_item.php]: change `or`

File: templates/beez3/html/com_newsfeeds/categories/default.php [hunk 4]
```diff
@@ -13,28 +13,26 @@
...
-		 <div class="category-desc base-desc">
-			<?php echo  JHtml::_('content.prepare', $this->params->get('categories_description'), '', 'com_newsfeeds.categories'); ?>
-			</div>
-		<?php  else: ?>
-			<?php //Otherwise get one from the database if it exists. ?>
-			<?php  if ($this->parent->description) : ?>
-				<div class="category-desc  base-desc">
-					<?php  echo JHtml::_('content.prepare', $this->parent->description, '', 'com_newsfeeds.categories'); ?>
+		<?php if ($this->params->get('show_base_description')) : ?>
+		<?php // If there is a description in the menu parameters use that; ?>
```
Comment (Quy) [inline templates/beez3/html/com_newsfeeds/categories/default.php]: remove `;`

File: templates/beez3/html/com_newsfeeds/categories/default.php [hunk 5]
```diff
@@ -13,28 +13,26 @@
...
-			<?php  endif; ?>
-		<?php  endif; ?>
-	<?php endif; ?>
-<?php
-echo $this->loadTemplate('items');
-?>
+			<?php else: ?>
+				<?php // Otherwise get one from the database if it exists. ?>
+				<?php if ($this->parent->description) : ?>
+					<div class="category-desc  base-desc">
```
Comment (Quy) [inline templates/beez3/html/com_newsfeeds/categories/default.php]: remove extra space

File: templates/beez3/html/com_newsfeeds/category/default_children.php [hunk 6]
```diff
@@ -33,9 +33,9 @@
 					<?php echo JHtml::_('content.prepare', $child->description, '', 'com_newsfeeds.category'); ?>
 				</div>
 			<?php endif; ?>
-            <?php endif; ?>
+			<?php endif; ?>
 
-            <?php if ($this->params->get('show_cat_items') == 1) :?>
+			<?php if ($this->params->get('show_cat_items') == 1) :?>
```
Comment (Quy) [inline templates/beez3/html/com_newsfeeds/category/default_children.php]: add space after `:`

File: templates/beez3/html/com_newsfeeds/categories/default_items.php [hunk 7]
```diff
@@ -31,7 +31,7 @@
 				<?php echo JHtml::_('content.prepare', $item->description, '', 'com_newsfeeds.categories'); ?>
 			</div>
 		<?php endif; ?>
-        <?php endif; ?>
+		<?php endif; ?>
 		<?php if ($this->params->get('show_cat_items_cat') == 1) :?>
```
Comment (Quy) [inline templates/beez3/html/com_newsfeeds/categories/default_items.php]: add space after `:`

File: templates/beez3/html/com_content/category/blog_item.php (hunk 2 above)
Comment (photodude) [reply templates/beez3/html/com_content/category/blog_item.php]: See this discussion on use of `and`/`or` vs `&&`/`||` in template mixed HTML/PHP files
there is no consensus on what we allow/disallow here.
https://github.com/joomla/coding-standards/pull/191

Comment (Quy) [reply templates/beez3/html/com_content/category/blog_item.php]: I understand. For this page, we have a mix of `and/or` and `&&/||` so it should be one or the other.

File: templates/beez3/html/com_newsfeeds/category/default_children.php (hunk 6 above)
Comment (photodude) [reply templates/beez3/html/com_newsfeeds/category/default_children.php]: fixed

File: templates/beez3/html/com_newsfeeds/categories/default_items.php (hunk 7 above)
Comment (photodude) [reply templates/beez3/html/com_newsfeeds/categories/default_items.php]: fixed

File: templates/beez3/html/com_newsfeeds/categories/default.php (hunk 5 above)
Comment (photodude) [reply templates/beez3/html/com_newsfeeds/categories/default.php]: fixed

File: templates/beez3/html/com_newsfeeds/categories/default.php (hunk 4 above)
Comment (photodude) [reply templates/beez3/html/com_newsfeeds/categories/default.php]: fixed

File: templates/beez3/html/com_contact/category/default_children.php (hunk 1 above)
Comment (photodude) [reply templates/beez3/html/com_contact/category/default_children.php]: fixed

File: templates/beez3/html/com_content/category/blog_item.php (hunk 2 above)
Comment (photodude) [reply templates/beez3/html/com_content/category/blog_item.php]: for now I'll set this file to `and`/`or`, we can always adjust it back using the auto fixers

Comment (Quy) [reply templates/beez3/html/com_content/category/blog_item.php]: My preference is `&&/||`. I still consider it PHP even though in HTML. I don't like that you have think/consider its syntax based on whether it is a PHP or HTML file.

Comment (photodude) [reply templates/beez3/html/com_content/category/blog_item.php]: That's my preference too. Code consistency for the win. but I made a choice here based on the some of the other arguments. 
https://github.com/joomla/coding-standards/pull/191#issuecomment-312276418

Comment (photodude) [reply templates/beez3/html/com_content/category/blog_item.php]: fixed despite the differing opinions on this. Code consistency for the win.

File: templates/beez3/html/com_content/category/blog_item.php (hunk 3 above)
Comment (photodude) [reply templates/beez3/html/com_content/category/blog_item.php]: fixed

File: templates/beez3/html/com_contact/category/default_children.php [hunk 8]
```diff
@@ -32,15 +32,15 @@
...
 			<?php endif; ?>
-            <?php endif; ?>
+			<?php endif; ?>
 
-            <?php if ($this->params->get('show_cat_items') == 1) :?>
-			<dl><dt>
-				<?php echo JText::_('COM_CONTACT_CAT_NUM'); ?></dt>
-				<dd><?php echo $child->numitems; ?></dd>
-			</dl>
+			<?php if ($this->params->get('show_cat_items') == 1) :?>
```
Comment (Quy) [inline templates/beez3/html/com_contact/category/default_children.php:37]: Add space after `:`

File: templates/beez3/html/com_contact/category/default_children.php [hunk 9]
```diff
@@ -32,15 +32,15 @@
...
+				<dl><dt>
+					<?php echo JText::_('COM_CONTACT_CAT_NUM'); ?></dt>
+					<dd><?php echo $child->numitems; ?></dd>
+				</dl>
 		<?php endif; ?>
-            <?php if (count($child->getChildren()) > 0 ) :
+			<?php if (count($child->getChildren()) > 0 ) :
```
Comment (Quy) [inline templates/beez3/html/com_contact/category/default_children.php:43]: Remove space after `0`

File: templates/beez3/html/com_contact/contact/default_form.php [hunk 10]
```diff
@@ -43,30 +43,30 @@
...
-			                            <?php endif; ?>
-			                         </div>
-			                         <div class="controls"><?php echo $field->input;?></div>
-			                    <?php endif;?>
-			                   </div>
-			               <?php endforeach;?>
-			          <?php endif ?>
-			     <?php endforeach;?>
+				<?php foreach ($this->form->getFieldsets() as $fieldset) : ?>
+					<?php if ($fieldset->name !== 'contact'):?>
```
Comment (Quy) [inline templates/beez3/html/com_contact/contact/default_form.php:49]: Add space before and after `:`

File: templates/beez3/html/com_contact/contact/default_form.php [hunk 11]
```diff
@@ -43,30 +43,30 @@
...
+						<?php $fields = $this->form->getFieldset($fieldset->name);?>
```
Comment (Quy) [inline templates/beez3/html/com_contact/contact/default_form.php:50]: Add space after `;`

File: templates/beez3/html/com_contact/contact/default_form.php [hunk 12]
```diff
@@ -43,30 +43,30 @@
...
+						<?php foreach ($fields as $field) : ?>
+							<div class="control-group">
+								<?php if ($field->hidden) : ?>
+									<div class="controls">
+									 <?php echo $field->input;?>
```
Comment (Quy) [inline templates/beez3/html/com_contact/contact/default_form.php:55]: Add space after `;`

File: templates/beez3/html/com_contact/contact/default_form.php [hunk 13]
```diff
@@ -43,30 +43,30 @@
...
+									</div>
+								<?php else:?>
+									<div class="control-label">
+										<?php echo $field->label; ?>
+										<?php if (!$field->required && $field->type !== 'Spacer') : ?>
+											<span class="optional"><?php echo JText::_('COM_CONTACT_OPTIONAL');?></span>
```
Comment (Quy) [inline templates/beez3/html/com_contact/contact/default_form.php:61]: Add space after `;`

File: templates/beez3/html/com_contact/contact/default_form.php [hunk 14]
```diff
@@ -43,30 +43,30 @@
...
+										<?php endif; ?>
+									</div>
+									<div class="controls"><?php echo $field->input;?></div>
```
Comment (Quy) [inline templates/beez3/html/com_contact/contact/default_form.php:64]: Add space after `;`

File: templates/beez3/html/com_contact/contact/default_form.php [hunk 15]
```diff
@@ -43,30 +43,30 @@
...
+								<?php endif;?>
```
Comment (Quy) [inline templates/beez3/html/com_contact/contact/default_form.php:65]: Same here

File: templates/beez3/html/com_contact/contact/default_form.php [hunk 16]
```diff
@@ -43,30 +43,30 @@
...
+							</div>
+						<?php endforeach;?>
```
Comment (Quy) [inline templates/beez3/html/com_contact/contact/default_form.php:67]: Same here

File: templates/beez3/html/com_contact/contact/default_form.php [hunk 17]
```diff
@@ -43,30 +43,30 @@
...
+					<?php endif ?>
+				<?php endforeach;?>
```
Comment (Quy) [inline templates/beez3/html/com_contact/contact/default_form.php:69]: Same here

File: templates/beez3/html/com_content/category/blog_item.php [hunk 18]
```diff
@@ -115,10 +115,10 @@
 		<?php echo JText::sprintf('COM_CONTENT_ARTICLE_HITS', $this->item->hits); ?>
 		</dd>
 <?php endif; ?>
-<?php if ($params->get('show_author') or $params->get('show_category') or $params->get('show_create_date') or $params->get('show_modify_date') or $params->get('show_publish_date') or $params->get('show_parent_category') or $params->get('show_hits')) :?>
- 	</dl>
+<?php if ($params->get('show_author') || $params->get('show_category') || $params->get('show_create_date') || $params->get('show_modify_date') || $params->get('show_publish_date') || $params->get('show_parent_category') || $params->get('show_hits')) :?>
```
Comment (Quy) [inline templates/beez3/html/com_content/category/blog_item.php:118]: Add space  after `:`

Comment (photodude) [reply templates/beez3/html/com_content/category/blog_item.php:118]: At some point we are deviating far from the scope of this PR. The intention was originally to only focus on fixing the indenting to use Tabs. 

Should we fix all of the other code style items? Sure, at some point. But we should be cautious in how much we add to the scope of this PR.

Comment (Quy) [reply templates/beez3/html/com_content/category/blog_item.php:118]: You're right. Lets do these in a separate PR.

Comment (photodude) [reply templates/beez3/html/com_content/category/blog_item.php:118]: Sounds like a plan.
Let's get this and #19350 merged and work towards fixing the rest of the CS items that can be auto fixed done. Then lets come back for the templates and see what can be done.

**Comment by zero-24** [comment]
Merged on review thanks @photodude

**Comment by photodude** [comment]
Thanks everyone

**Pull request merged**